packages/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -1,38 +1,441 @@
 import { SlotClassNames } from '@fluentui/react-utilities';
-import { makeStyles, mergeClasses } from '@griffel/react';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { SpinButtonSlots, SpinButtonState } from './SpinButton.types';
+import { tokens } from '@fluentui/react-theme';
+import { useInputStyles_unstable } from '@fluentui/react-input';
 
-/**
- * @deprecated Use `spinButtonClassNames.root` instead.
- */
-export const spinButtonClassName = 'fui-SpinButton';
 export const spinButtonClassNames: SlotClassNames<SpinButtonSlots> = {
   root: 'fui-SpinButton',
   input: 'fui-SpinButton__input',
-  incrementControl: 'fui-SpinButton__incrementControl',
-  decrementControl: 'fui-SpinButton__decrementControl',
+  incrementButton: 'fui-SpinButton__incrementButton',
+  decrementButton: 'fui-SpinButton__decrementButton',
 };
 
-/**
- * Styles for the root slot
- */
-const useStyles = makeStyles({
-  root: {
-    // TODO Add default styles for the root element
+const spinButtonExtraClassNames = {
+  button: 'fui-SpinButton__button',
+};
+
+// TODO(sharing) use theme values once available
+const horizontalSpacing = {
+  xs: '4px',
+};
+
+const useRootStyles = makeStyles({
+  base: {
+    display: 'inline-grid',
+    gridTemplateColumns: `1fr 24px`,
+    gridTemplateRows: '1fr 1fr',
+    columnGap: horizontalSpacing.xs,
+    rowGap: 0,
+    paddingRight: 0,
+    position: 'relative',
+    // Remove the border styles from react-input
+    ...shorthands.border('0'),
+    isolation: 'isolate',
+
+    // Apply border styles on the ::before pseudo element.
+    // We cannot use ::after since react-input uses that
+    // for the selector styles.
+    // Using the pseudo element allows us to place the border
+    // above content in the component which ensures the buttons
+    // line up visually with the border as expected. Without this
+    // there is a bit of a gap which can become very noticeable
+    // at high zoom or when OS zoom levels are not divisible by 2
+    // (e.g., 150% on Windows in Firefox)
+    // This is most noticeable on the "outline" appearance which is
+    // also the default so it feels worth the extra ceremony to get right.
+    '::before': {
+      content: '""',
+      boxSizing: 'border-box',
+      position: 'absolute',
+      top: '-1px',
+      right: '-1px',
+      bottom: '-1px',
+      left: '-1px',
+      ...shorthands.borderRadius(tokens.borderRadiusMedium),
+      pointerEvents: 'none',
+      zIndex: 10,
+    },
+
+    '::after': {
+      zIndex: 20,
```

Comment:
I feel like we need tokens for semantic `z-index` values 😅

Reply:
Yes, fortunately the [`isolation` property](https://developer.mozilla.org/en-US/docs/Web/CSS/isolation) let's us easily create a new stacking context.